### PR TITLE
feat(uiprojector): decouple projector identity from render type

### LINKED ISSUE #558: Decouple projector identity from render type in uiprojector
### Current Limitation

In `backend/pkg/uiprojector`, the `Projector` interface conflates two distinct concerns into a single `Type() ProjectorType` method:

1. **Registry key** — used by the assembler to look up which projector handles a given `SectionBlueprint.Projector` value (`assembler.go:64`).
2. **Render/output type** — copied verbatim into `Section.Type` at `assembler.go:91`, which the frontend uses to decide how to display the content.

Because these two concerns share one field:

- Two projectors that process input differently but emit the same render shape cannot coexist (e.g. a `JSONSchemaFormProjector` and a `YAMLFormProjector` both wanting `Section.Type = "FORM"` would collide in `NewAssembler`'s duplicate check).
- A single projector cannot emit different render types based on runtime data. A concrete case: a `PaymentProjector` needs to return a `REDIRECT` section for online gateways and a `DESCRIPTION` section for offline payment instructions — the choice depends on the payment method configuration, not on the projector identity.

### Suggested Improvement

Split the two concerns:

- Keep `Projector.Type() ProjectorType` as the **registry key only** — matches `SectionBlueprint.Projector`, used solely for lookup.
- Change `Project(ctx, template, data) (any, error)` to return a struct that bundles the render type with the content, e.g.:

  ```go
  type Projection struct {
      Type    SectionType
      Content any
  }

  Project(ctx context.Context, templateContent []byte, data any) (Projection, error)
  ```

  The assembler then copies `Projection.Type` into `Section.Type` and `Projection.Content` into `Section.Content`.

This lets the projector decide the render type per invocation based on runtime data, and allows multiple distinct projectors to emit the same render type without collision.

### Version

main @ a79742e7

### Additional Context

Discovered while designing a `PaymentProjector` that needs to dispatch between online (redirect) and offline (description) payment flows. Worth considering alongside this change whether the blueprint should be able to constrain the set of render types a section may emit per zone, though that is not required for the immediate use case.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

`Projector.Type()` was doing double duty as both the registry key and the render type written to `Section.Type`. That coupling blocked two real cases: (a) multiple distinct projectors that legitimately emit the same render shape, and (b) a single projector that needs to choose its render type at runtime (e.g. a `PaymentProjector` returning `REDIRECT` for online gateways vs `DESCRIPTION` for offline instructions).

This PR splits the two concerns: `Type()` stays as the registry key, and `Project` now returns a `Projection{Type, Content}` so the render type is chosen per invocation.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

The `Projector` interface signature changed, which is a breaking change for any external implementer. The package currently has no callers outside its own tests, so production impact is nil.

## Changes Made

- Added `Projection{Type SectionType, Content any}` in `pkg/uiprojector/projector.go`.
- Changed `Projector.Project` to return `(Projection, error)` instead of `(any, error)`.
- `Type()` is now documented as registry-key only; render type is read from `Projection.Type`.
- Updated `FormProjector`, `MarkdownProjector`, `RawProjector` to emit the new struct, using new `SectionTypeForm` / `SectionTypeMarkdown` / `SectionTypeRaw` constants added in `defaults.go`.
- `Assembler.Assemble` now copies `projection.Type` and `projection.Content` into the resulting `Section` (previously it copied `proj.Type()` into `Section.Type`).
- Updated unit tests and the integration test stub to match the new contract.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

`go test ./pkg/uiprojector/...` passes (both unit and integration suites).

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

`pkg/uiprojector/README.md` was not updated in this PR — worth a follow-up once the `PaymentProjector` lands and we can document the dynamic-render-type pattern with a real example.

## Related Issues

Closes #558

## Additional Notes

Reviewer note on what this does **not** address (called out in #558's discussion but intentionally left out of scope):

- Stable section ordering — `Blueprint.Sections` is still a map; ordering is unaddressed here.
- Partial-success on projection errors — one bad section still aborts the whole `Assemble`.
- Template caching — the existing TODO in `assembler.go` is unchanged.
- Constraining which render types a zone may accept — possible follow-up if dynamic projectors proliferate.

## Deployment Notes

No runtime behavior changes for existing call sites because the package has no callers outside its own tests yet. No migrations or config changes required.

### CODE REVIEW

File: backend/pkg/uiprojector/assembler.go [hunk 1]
```diff
@@ -81,16 +81,16 @@ func (a *Assembler) Assemble(ctx context.Context, blueprint *Blueprint, facts Fa
...
 		}
 
 		sections[zone] = Section{
 			ID:      sb.ID,
-			Type:    SectionType(proj.Type()),
+			Type:    projection.Type,
 			Title:   sb.Title,
-			Content: content,
+			Content: projection.Content,
 		}
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply backend/pkg/uiprojector/assembler.go:97]: Addressed.

**Review by ginaxu1 — APPROVED** [review]

**Review by sthanikan2000 — APPROVED** [review]
